smart-contracts/test/Lock/execTransaction.js

```diff
@@ -0,0 +1,79 @@
+const {
+  purchaseKey,
+  reverts,
+  deployERC20,
+  deployLock,
+  deployContracts,
+} = require('../helpers')
+const { ethers } = require('hardhat')
+
+let unlock
+let lock
+
+contract('Lock / execTransaction', (accounts) => {
+  const [deployer, lockOwner, keyOwner, spender, other] = accounts
+  let token
+
+  before(async () => {
```

Comment:
Can you add test to make sure the lock can "spend" its funds (both Eth and ERC20)?

Reply:
cc @clemsos ^^

Reply:
yes, added tests for ERC20 spending. As we still have `withdraw`, I dont think we should make this function support ETH transfer as it will add complexity to it for duplicating what already supported

Reply:
It's just a test! We should be much better at identifying regressions. 
We meed a test to make sure that "spending" eth works with `execTransaction`. Right now we don;t have any!